packages/next/src/build/webpack/loaders/lightningcss-loader/src/loader.ts

```diff
@@ -0,0 +1,308 @@
+        }
+        const newUrl = prefix ? `${prefix}!${url}` : url
+        let importName = importUrlToNameMap.get(newUrl)
+        if (!importName) {
+          importName = `___CSS_LOADER_AT_RULE_IMPORT_${importUrlToNameMap.size}___`
+          importUrlToNameMap.set(newUrl, importName)
+
+          const importUrl = visitorOptions.urlHandler(newUrl)
+          imports.push({
+            type: 'rule_import',
+            importName,
+            url: importUrl,
+          })
+        }
+        apis.push({ importName, media })
+        // Bug of lightningcss
+        return { type: 'ignored', value: '' }
+      },
+    },
+    Url(node) {
+      return handleUrl(node)
+    },
+  }
+}
+
+const LOADER_NAME = `lightningcss-loader`
+export async function LightningCssLoader(
+  this: LoaderContext<ILightningCssLoaderConfig>,
+  source: string,
+  prevMap?: Record<string, any>
+): Promise<void> {
+  const done = this.async()
+  const options = this.getOptions()
+  const { implementation, targets: userTargets, ...opts } = options
+
+  options.modules ??= {}
+
+  if (implementation && typeof implementation.transformCss !== 'function') {
+    done(
+      new TypeError(
+        `[${LOADER_NAME}]: options.implementation.transformCss must be an 'lightningcss' transform function. Received ${typeof implementation.transformCss}`
+      )
+    )
+    return
+  }
+
+  const exports: CssExport[] = []
+  const imports: CssImport[] = []
+  const apis: ApiParam[] = []
+  const replacements: ApiReplacement[] = []
+
+  if (options.modules?.exportOnlyLocals !== true) {
+    imports.unshift({
+      type: 'api_import',
+      importName: '___CSS_LOADER_API_IMPORT___',
+      url: stringifyRequest(
+        this,
+        require.resolve('../../css-loader/src/runtime/api')
+      ),
+    })
+  }
+  const transform = implementation?.transformCss ?? transformCss
+
+  const urlResolver = this.getResolve({
+    conditionNames: ['asset'],
+    mainFields: ['asset'],
+    mainFiles: [],
+    extensions: [],
+  })
+
+  const replacedUrls = new Map<number, string>()
+
+  try {
+    const {
+      code,
+      map,
+      exports: moduleExports,
+    } = transform({
+      visitor: createVisitor(
+        {
+          urlHandler: (url) =>
+            stringifyRequest(
+              this,
+              getPreRequester(this)(options.importLoaders ?? 0) + url
+            ),
+          urlFilter: getFilter(options.url, this.resourcePath),
+          importFilter: getFilter(options.import, this.resourcePath),
+
+          context: this.context,
+          urlResolver,
+        },
+        apis,
+        imports,
+        replacements,
+        replacedUrls
+      ),
+      cssModules: options.modules
+        ? {
+            pattern: process.env.__NEXT_TEST_MODE
+              ? '[name]__[local]'
+              : '[name]__[hash]__[local]',
+          }
+        : undefined,
+      filename: this.resourcePath,
+      code: Buffer.from(source),
```

Comment:
`node:buffer` is a Node.js-only API. While `Buffer` is a subclass of `Uint8Array`, we can replace `Buffer.from` with `TextEncoder`:

```ts
const encoder = new TextEncoder();

export async function LightningCssLoader(/* ... */) {
  // ......
  code: encoder.encode(source)
  // ......
}
```